src/System.Diagnostics.Process/tests/ProcessTests.Linux.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Security;
+using Xunit;
+using Xunit.NetCore.Extensions;
+
+namespace System.Diagnostics.Tests
+{
+    public partial class ProcessTests : ProcessTestBase
+    {
+        // expected behavior in Linux is a bit unknown: 
+        // If non of the programs are installed it uses shell. So result depends on what's installed
+        // we don't want to depend on linux build agents that necessarily have certain programs installed. 
+        // the test expectations need to be set carefully
+        
+        [Fact]
+        public void TestWithFilename_OpensASilentConsoleApp_AppShouldPrintOrReturnSomething()
+        {
+            // TODO
+            // open a silent project. i.e a console app that only prints something to the console, or returns a value, or sets exit code to an specific number
+        }
+
+        [Fact]
+        [OuterLoop("Launches default application")]
+        public void TestWithFilename_ShouldUseOpenWithDefaultApp()
+        {
+            // maybe pass down to open current file
+            // TODO
+            // string filePath; // get executing assembly path and then decide on opening a certain file from that path
+            //using (var p = Process.Start("/usr/bin/xdg-open", filePath));
+            //{
+            //Thread.Sleep(500);
+            //Assert.False(px.HasExited);
+            //px.Kill();
+            //px.WaitForExit();
+            //Assert.True(px.HasExited);
+            //}
+        }
+
+        [Fact]
+        [OuterLoop("Launches default browser")]
+        public void TestWithUrl_ShouldUseOpenWithDefaultApp()
+        {
+            using (var p = Process.Start("/usr/bin/xdg-open", "http://www.google.com"))
+            {
+                Thread.Sleep(500);
+                Assert.False(px.HasExited);
+                px.Kill();
+                px.WaitForExit();
+                Assert.True(px.HasExited);
+            }
+        }
+
+        [Fact]
+        public void UseShellExecuteIsTrue_ExitCodeReturnValueIsDifferentAfterMyNewChanges()
+        {
+            using (var p = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = "exit", Arguments = "42" }))
+            {
+                Assert.True(p.WaitForExit(WaitInMS));
+//                Assert.Equal(42, p.ExitCode); // Assert.Equal Failure with after adding my changes: Expected: 42, Actual: 1
+            }
+        }
+
+        [Fact]
+        public void UseShellExecuteIsFalse_ThrowsWithOrWithoutMyChanges()
+        {
+            // The test below fails on linux with or without my changes 
```

Comment:
Observation 1: the commented out test below fails on linux with or without my changes.
Notice that UseShellExecute is false.